sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateAlertDao.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey, Mario Caccamo @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+package uk.ac.bbsrc.tgac.miso.persistence.impl;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.criterion.Restrictions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.eaglegenomics.simlims.core.store.SecurityStore;
+
+import net.sf.ehcache.CacheManager;
+
+import uk.ac.bbsrc.tgac.miso.core.event.Alert;
+import uk.ac.bbsrc.tgac.miso.core.event.impl.DefaultAlert;
+import uk.ac.bbsrc.tgac.miso.core.event.type.AlertLevel;
+import uk.ac.bbsrc.tgac.miso.core.store.AlertStore;
+import uk.ac.bbsrc.tgac.miso.core.util.CoverageIgnore;
+
+/**
+ * uk.ac.bbsrc.tgac.miso.hibernate.persistence.impl
+ *
+ * @author Heather Armstrong
+ * @since 0.2.43
+ */
+
+@Transactional(rollbackFor = Exception.class)
+public class HibernateAlertDao implements AlertStore {
+
+  protected static final Logger log = LoggerFactory.getLogger(HibernateAlertDao.class);
+
+  private SecurityStore securityDao;
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  @Autowired
+  private JdbcTemplate template;
+
+  @Autowired
+  private CacheManager cacheManager;
+
+  @CoverageIgnore
```

Comment:
Heh I made a mental note during your presentation to check this again, looks like you beat me to it!